yt_dlp/extractor/triller.py

```diff
@@ -0,0 +1,309 @@
+import itertools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    format_field,
+    int_or_none,
+    str_or_none,
+    traverse_obj,
+    unified_strdate,
+    unified_timestamp,
+    url_basename,
+    ExtractorError,
+    LazyList,
+)
+
+
+class TrillerBaseIE(InfoExtractor):
+    _NETRC_MACHINE = 'triller'
+    _AUTH_TOKEN = None
+    _API_BASE_URL = 'https://social.triller.co/v1.5'
+
+    def _perform_login(self, username, password):
+        if self._AUTH_TOKEN:
+            return
+
+        user_check = self._download_json(
+            f'{self._API_BASE_URL}/api/user/is-valid-username', None, note='Logging in',
+            fatal=False, expected_status=400, headers={
+                'Content-Type': 'application/json',
+                'Origin': 'https://triller.co',
+            }, data=json.dumps({'username': username}, separators=(',', ':')).encode('utf-8'))
+        if user_check.get('status'):  # endpoint returns "status":false if username exists
+            raise ExtractorError('Unable to login: Invalid username', expected=True)
+
+        credentials = {
+            'username': username,
+            'password': password,
+        }
+        login = self._download_json(
+            f'{self._API_BASE_URL}/user/auth', None, note=False, fatal=False,
+            expected_status=400, headers={
+                'Content-Type': 'application/json',
+                'Origin': 'https://triller.co',
+            }, data=json.dumps(credentials, separators=(',', ':')).encode('utf-8'))
+        if not login.get('auth_token'):
+            if login.get('error') == 1008:
+                raise ExtractorError('Unable to login: Incorrect password', expected=True)
+            raise ExtractorError('Unable to login')
+
+        self._AUTH_TOKEN = login['auth_token']
+
+    def _get_comments(self, video_id, limit=15):
+        comment_info = self._download_json(
+            f'{self._API_BASE_URL}/api/videos/{video_id}/comments_v2',
+            video_id, fatal=False, note=False, errnote='Unable to extract comments',
+            headers={'Origin': 'https://triller.co'}, query={'limit': limit}) or {}
+        if not comment_info.get('comments'):
+            return
+        yield [{
```

Comment:
```suggestion
        return [{
```

Reply:
--write-comments throws an error with return instead of yield ?